warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -2796,6 +2797,20 @@ public Tuple2<CloseableIterable<QueryPlan>,Boolean> getQueryRanges(ScannerFactor
         return new Tuple2<>(ranges, needsFullTable);
     }
 
+    protected ASTJexlScript timedSortQueryBeforeGlobalIndex(ShardQueryConfiguration config, MetadataHelper metadataHelper) throws DatawaveQueryException {
+        return visitorManager.timedVisit(config.getTimers(), "SortQueryBeforeGlobalIndex", () -> {
```

Comment:
I am missing if there is a test that covers this change.  Is there a way to test that it did what we want?  Maybe testing the final plan to see that is is different when isSortQueryBeforeGlobalIndex is set to true?

Reply:
Added a test to verify the query terms are re-ordered by field cardinality and discovered another problem with the RebuildingScannerHelper which led to it's removal from a test.